### PR TITLE
Whisper: Support ORT 1.17.1

### PR DESCRIPTION
**Pull request by jambayk**
## Describe your changes
ORT 1.17.1 made some changes to the beam search ops. For version 1.17.1, the beam search op used will be `WhisperBeamSearch`. 

Fixes:
#1014

## Checklist before requesting a review
- [ ] Add unit tests for this change.
- [ ] Make sure all tests can pass.
- [ ] Update documents if necessary.
- [ ] Lint and apply fixes to your code by running `lintrunner -a`
- [ ] Is this a user-facing change? If yes, give a description of this change to be included in the release notes.
- [ ] Is this PR including examples changes? If yes, please remember to update [example documentation](https://github.com/microsoft/Olive/blob/main/docs/source/examples.md) in a follow-up PR.

### CODE REVIEW

**Comment by xiaoyu-work** [comment]
I'm not sure if this is the root cause, cause I'm able to run `test_transcription.py` with ORT 1.17.1 successfully without `--predict_timestamps`. Do you know the reason?

**Comment by jambayk** [comment]
> I'm not sure if this is the root cause, cause I'm able to run `test_transcription.py` with ORT 1.17.1 successfully without `--predict_timestamps`. Do you know the reason?

@xiaoyu-work I am not entirely sure about the root cause but you can see in this PR https://github.com/microsoft/onnxruntime/pull/19509 that some token ids are not part of the model attributes and not inferred using offsets. This affects the logits processor which does the time stamp prediction. I think the lack of these values in the beam search node's attributes causes the issue.

**Comment by xiaoyu-work** [comment]
> > I'm not sure if this is the root cause, cause I'm able to run `test_transcription.py` with ORT 1.17.1 successfully without `--predict_timestamps`. Do you know the reason?
> 
> @xiaoyu-work you can see in this PR [microsoft/onnxruntime#19509](https://github.com/microsoft/onnxruntime/pull/19509) that some token ids are now part of the model attributes and not inferred using offsets. This affects the logits processor which does the time stamp prediction. I think the lack of these values in the beam search node's attributes causes the issue.

Can you test if this change works?

**Comment by jambayk** [comment]
> > > I'm not sure if this is the root cause, cause I'm able to run `test_transcription.py` with ORT 1.17.1 successfully without `--predict_timestamps`. Do you know the reason?
> > 
> > 
> > @xiaoyu-work you can see in this PR [microsoft/onnxruntime#19509](https://github.com/microsoft/onnxruntime/pull/19509) that some token ids are now part of the model attributes and not inferred using offsets. This affects the logits processor which does the time stamp prediction. I think the lack of these values in the beam search node's attributes causes the issue.
> 
> Can you test if this change works?

It's already tested.

**Review by xiaoyu-work — DISMISSED** [review]

File: olive/passes/onnx/insert_beam_search.py [hunk 1]
```diff
@@ -84,9 +98,20 @@ def chain_model(
...
+        version_1_17_1 = version.parse(OrtVersion) >= version.parse("1.17.1")
+        # NOTE: will ignore cross qk related options for now
+
+        # get tokenizer
+        model_name = options["model_name"] or model_config.get("_name_or_path")
+        if not model_name:
+            raise ValueError("model_name is not provided and cannot be inferred from model attributes.")
+        if not options["model_name"]:
+            logger.info("Inferred model name from model attributes: %s", model_name)
+        tokenizer = AutoTokenizer.from_pretrained(model_name)
```
Comment (xiaoyu-work) [inline olive/passes/onnx/insert_beam_search.py:114]: Nit: seems only ORT >= 1.17.1 needs load tokennizer. Add a condition check here can save unnecessary load.

Comment (jambayk) [reply olive/passes/onnx/insert_beam_search.py]: Thanks! updated to only get tokenizer when needed.

**Review by xiaoyu-work — DISMISSED** [review]

File: olive/passes/onnx/insert_beam_search.py [hunk 2]
```diff
@@ -21,14 +21,21 @@
...
-    """Insert Beam Search Op."""
+    """Insert Beam Search Op. Uses WhisperBeamSearch op if ORT version >= 1.17.1, else uses BeamSearch op."""
 
     _accepts_composite_model = True
 
     @classmethod
     def _default_config(cls, accelerator_spec: AcceleratorSpec) -> Dict[str, PassConfigParam]:
         use_gpu = accelerator_spec.accelerator_type == Device.GPU
         config = {
+            "model_name": PassConfigParam(
```
Comment (guotuofeng) [inline olive/passes/onnx/insert_beam_search.py:32]: could the model_name be get from model_config or hf_config?

Comment (jambayk) [reply olive/passes/onnx/insert_beam_search.py:32]: yes, you can see in the description that the model_name will be inferred from the model_config is not provided. I kept this here in case they user has a model with custom tokenization and want to use it.

Comment (guotuofeng) [reply olive/passes/onnx/insert_beam_search.py:32]: for custom tokenizer case, should we provide the function to load the tokenizer in user_script? customized tokenizer might not be supported by  AutoTokenizer.from_pretrained

Comment (guotuofeng) [reply olive/passes/onnx/insert_beam_search.py:32]: From my understanding, AutoTokenizer.from_pretrained will mostly be the model supported by Hugging Face. Is that the case?

Comment (jambayk) [reply olive/passes/onnx/insert_beam_search.py:32]: the tokenizer with the same. It is just that they might have changed the tokenizer config json when saving the model. so as long as they provide the correct model name or path here, it will be okay.

Comment (guotuofeng) [reply olive/passes/onnx/insert_beam_search.py:32]: so we will not support https://github.com/BlinkDL/ChatRWKV/blob/main/tokenizer/rwkv_tokenizer.py?

Comment (jambayk) [reply olive/passes/onnx/insert_beam_search.py:32]: i was not clear, I meant if they changed the vocabulary which would in turn change the tokenization.

Comment (jambayk) [reply olive/passes/onnx/insert_beam_search.py:32]: the use case is similar to `model_name` here https://github.com/microsoft/Olive/blob/main/olive/passes/utils/whisper_prepost.py#L14. This is to support locally save fine-tuned hf whisper models that might have updated vocab, etc. Not to support a custom model or tokenizer class.

Comment (guotuofeng) [reply olive/passes/onnx/insert_beam_search.py:32]: for whisper, I am suppose the model_name will not be used?

Comment (jambayk) [reply olive/passes/onnx/insert_beam_search.py]: updated to remove as discussed offline.

**Review by guotuofeng — APPROVED** [review]
